src/recorder.rs

```diff
@@ -0,0 +1,283 @@
+        }
+    };
+
+    // open destination file
+    let destination: std::fs::File = {
+        match std::fs::File::create(path.clone()) {
+            Ok(f) => f,
+            Err(error) => {
+                eprintln!("could not open destination: {:?}\n{error}", path);
+                std::process::exit(1);
+            }
+        }
+    };
+
+    // open temporary (intermediate msgpack) file
+    let mut temp_path = path.clone();
+    temp_path.pop();
+    let temporary = match tempfile_in(temp_path.clone()) {
+        Ok(t) => t,
+        Err(error) => {
+            eprintln!("could not open temporary file in: {:?}\n{error}", temp_path);
+            std::process::exit(1);
+        }
+    };
+
+    // configure debug log
+    let debug_output: Box<dyn Output> = Box::new(Stderr::new());
+
+    let level = match matches.get_count("VERBOSE") {
+        0 => Level::Info,
+        1 => Level::Debug,
+        _ => Level::Trace,
+    };
+
+    let debug_log = if level <= Level::Info {
+        LogBuilder::new().format(ringlog::default_format)
+    } else {
+        LogBuilder::new()
+    }
+    .output(debug_output)
+    .build()
+    .expect("failed to initialize debug log");
+
+    let mut log = MultiLogBuilder::new()
+        .level_filter(level.to_level_filter())
+        .default(debug_log)
+        .build()
+        .start();
+
+    // initialize async runtime
+    let rt = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .worker_threads(1)
+        .thread_name("rezolus")
+        .build()
+        .expect("failed to launch async runtime");
+
+    // spawn logging thread
+    rt.spawn(async move {
+        loop {
+            tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+            let _ = log.flush();
+        }
+    });
+
+    ctrlc::set_handler(move || {
+        if RUNNING.load(Ordering::SeqCst) {
+            eprintln!("finalizing recording... please wait...");
+            RUNNING.store(false, Ordering::SeqCst);
+        } else {
+            eprintln!("terminating...");
+            std::process::exit(2);
+        }
+    })
+    .expect("failed to set ctrl-c handler");
+
+    // spawn recorder thread
+    rt.block_on(async move {
+        recorder(addr, destination, temporary, interval).await;
+    });
+}
+
+async fn recorder(
+    addr: SocketAddr,
+    destination: std::fs::File,
+    temporary: std::fs::File,
+    interval: Duration,
+) {
+    let mut temporary = tokio::fs::File::from_std(temporary);
+
+    let mut interval = tokio::time::interval(interval);
+
+    let mut client = None;
+
+    while RUNNING.load(Ordering::Relaxed) {
+        if client.is_none() {
+            debug!("connecting to Rezolus...");
+
+            match TcpStream::connect(addr).await {
+                Ok(s) => {
+                    if s.set_nodelay(true).is_err() {
+                        continue;
+                    }
+
+                    debug!("performing http2 handshake...");
```

Comment:
Curious if there is actually a benefit for HTTP/2 here given that we aren't pipe-lining or worried about head of line blocking. Better compression?

Reply:
Probably no real benefit, but we know Rezolus supports H2, so I don't see a downside to using it.